plugins/modules/lvm_pv.py

```diff
@@ -0,0 +1,190 @@
+    cmd = ['pvs', '--noheadings', '--nosuffix', '--units', 'b', '-o', 'pv_size', device]
+    rc, out, err = module.run_command(cmd, check_rc=True)
+    return int(out.strip())
+
+
+def rescan_device(module, device):
+    """Perform storage rescan for the device."""
+    # Extract the base device name (e.g., /dev/sdb -> sdb)
+    base_device = os.path.basename(device)
+    rescan_path = "/sys/block/{0}/device/rescan".format(base_device)
+
+    if os.path.exists(rescan_path):
+        try:
+            with open(rescan_path, 'w') as f:
+                f.write('1')
+            return True
+        except IOError as e:
+            module.warn("Failed to rescan device {0}: {1}".format(device, str(e)))
+            return False
+    else:
+        module.warn("Rescan path not found for device {0}".format(device))
+        return False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            device=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            force=dict(type='bool', default=False),
+            resize=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+
+    device = module.params['device']
+    state = module.params['state']
+    force = module.params['force']
+    resize = module.params['resize']
+    changed = False
+    actions = []
+
+    # Validate device existence for present state
+    if state == 'present' and not os.path.exists(device):
+        module.fail_json(msg="Device %s not found" % device)
+
+    is_pv = get_pv_status(module, device)
+
+    if state == 'present':
+        # Create PV if needed
+        if not is_pv:
+            if module.check_mode:
+                changed = True
+                actions.append('created')
+            else:
+                cmd = ['pvcreate']
+                if force:
+                    cmd.append('-f')
+                cmd.append(device)
+                rc, out, err = module.run_command(cmd, check_rc=True)
+                changed = True
+                actions.append('created')
+            is_pv = True
+
+        # Handle resizing
+        if resize and is_pv:
+            if not module.check_mode:
+                # Perform device rescan if each time
+                if rescan_device(module, device):
+                    actions.append('rescanned')
+                original_size = get_pv_size(module, device)
+                rc, out, err = module.run_command(['pvresize', device], check_rc=True)
+                new_size = get_pv_size(module, device)
+                if new_size != original_size:
+                    changed = True
+                    actions.append('resized')
+            else:
+                # In check mode, assume resize would change
+                changed = True
+                actions.append('would be resized')
+
+    elif state == 'absent':
+        if is_pv:
+            if module.check_mode:
```

Comment:
I think this should be:
```suggestion
            if not module.check_mode:
```
no? Or you are missing the check mode statements and an `else` (like lines 139-142).

Reply:
Hi @russoz, yes you are right, I missed the `else`  block. Which I added below.

Regarding the tests, is there anybody else who can help out with those?
I will be quite busy in the following weeks and do not think will be able to add them.

Thank you for your support!